src/manager/github_runner_manager.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Client for managing self-hosted runner on GitHub side."""
+
+from enum import Enum
+from typing import Sequence
+
+from charm_state import GithubPath
+from github_client import GithubClient
+from github_type import GitHubRunnerStatus, SelfHostedRunner
+
+
+class GithubRunnerState(str, Enum):
+    """State of the self-hosted runner on GitHub.
+
+    Attributes:
+        BUSY: Runner is working on a job assigned by GitHub.
+        IDLE: Runner is waiting to take a job.
+        OFFLINE: Runner is not connected to GitHub.
+    """
+
+    BUSY = "busy"
+    IDLE = "idle"
+    OFFLINE = "offline"
+
+    @staticmethod
+    def from_runner(runner: SelfHostedRunner) -> "GithubRunnerState":
+        """Construct the object from GtiHub runner information.
+
```

Comment:
nit: There's no official documentation about the statuses, but I've found this: 
```suggestion
        There is no official documentation on statuses, refer to source code statuses:
        
 
 https://github.com/actions/runner/blob/a77fe8a53fe9fc2a77ae4f511d46d04197bdde66/src/Sdk/DTWebApi/WebApi/TaskAgentStatus.cs#L1
```

Reply:
As discussed in our meeting, these status does not reflect the GitHub API return value well.